roles/openshift_health_checker/openshift_checks/memory_availability.py

```diff
@@ -0,0 +1,31 @@
+# pylint: disable=missing-docstring
+from openshift_checks import OpenShiftCheck, OpenShiftCheckException, get_var
+
+
+class MemoryAvailability(OpenShiftCheck):
+    """Check that recommended memory is available."""
+
+    name = "memory_availability"
+    tags = ["preflight"]
+
+    recommended_memory_mb = {
+        "nodes": 8 * 1024,
+        "masters": 16 * 1024,
+        "etcd": 10 * 1024,
+    }
+
```

Comment:
We need to implement `is_active` for this check, similar to DiskAvailability: only run for hosts that we actually have a recommendation to verify.